### PR TITLE
  Diatomic Morse (for exercises)

### PR DESCRIPTION
**Pull request by mahrossi**
Modification to driver so that exercises I am doing become easier.

### CODE REVIEW

File: drivers/f90/pes/morse.f [hunk 1]
```diff
@@ -7,4 +7,16 @@ SUBROUTINE getmorse(r0,D,a,q,pot,force)
...
+      SUBROUTINE getmorsedia(r0,D,a,q,pot,force)
+         IMPLICIT NONE
+         DOUBLE PRECISION r0, D, a, q(2,3), pot, force(2,3)
+         DOUBLE PRECISION nq, dr, diff(3)
+         diff(:)=(q(1,:)-q(2,:))
+         nq=dsqrt(dot_product(diff,diff))
+         dr=nq-r0
+         pot=D*(dexp(-2.*a*dr)-2.*dexp(-a*dr))
+         force(1,:)=2*a*D*(dexp(-a*dr)-dexp(-2.0*a*dr))*diff/nq
+         force(2,:)=2*a*D*(dexp(-a*dr)-dexp(-2.0*a*dr))*diff/nq
```
Comment (litman90) [inline drivers/f90/pes/morse.f:21]: Could you double-check? I think there is a minus missing
As reference, the 1D morse is:
      SUBROUTINE getmorse(r0,D,a,q,pot,force)
         IMPLICIT NONE
         DOUBLE PRECISION r0, D, a, q(1,3), pot, force(1,3)
         DOUBLE PRECISION nq, dr
         nq=dsqrt(sum(q*q))
         dr=nq-r0
         pot=D*(dexp(-2.*a*dr)-2.*dexp(-a*dr))
         force=-2*a*D*(dexp(-a*dr)-dexp(-2.0*a*dr))*q/nq
      END SUBROUTINE

**Review by litman90 — CHANGES_REQUESTED** [review]

File: drivers/f90/pes/morse.f (hunk 1 above)
Comment (mahrossi) [reply drivers/f90/pes/morse.f:21]: it is implicit in the "diff" term (that does not exist in the 1D case). I can change for clarity, but before pushing I checked which sign combination would make the simulation stable ;-) the other case explodes, so ... - anyways, we can merge later when I finished all the exercises

Comment (mahrossi) [reply drivers/f90/pes/morse.f:21]: EDIT: actually well spotted - the version I pushed was not the version I was running with. All should be fine now. As mentioned, beyond the math all checks pass - dynamics are stable, geometry optimization provides expected geometry and vibrations provide expected frequency.

Comment (litman90) [reply drivers/f90/pes/morse.f:21]: cool, thanks for checking

**Review by litman90 — APPROVED** [review]

**Comment by litman90** [comment]
@mahrossi I merged it, please delete the branch at your earliest convenience.

**Pull request merged**